subsystems/provisioning/networkmanager.go

```diff
@@ -737,6 +739,20 @@ func (w *Provisioning) mainLoop(ctx context.Context) {
 			}
 		}
 
+		offlineRebootTimeout := w.cfg.DeviceRebootAfterOfflineMinutes > 0 &&
+			lastConnectivity.Before(now.Add(time.Duration(w.cfg.DeviceRebootAfterOfflineMinutes)*-1))
+		if offlineRebootTimeout {
+			w.logger.Infof("device has been offline for more than %s minutes, rebooting", w.cfg.DeviceRebootAfterOfflineMinutes)
+
+			syscall.Sync() // flush file system buffers
+			cmd := exec.Command("systemctl", "reboot")
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				w.logger.Error(errw.Wrapf(err, "running 'systemctl reboot' %s", output))
+			}
+			time.Sleep(time.Second * 100) // systemd DefaultTimeoutStopSec defaults to 90 seconds
```

Comment:
Can't do this, or agent itself won't shut down when called and will just hang until the sleep is done. It will also fail healthchecks if the loop pauses here for too long. There's a function for pausing the loop though. See line 623 above for the previous use of the following:

```
if !w.mainLoopHealth.Sleep(ctx, time.Second*100) {
	return
}
w.logger.Errorf("Failed to reboot after %s time!", time.Second * 100)			
```

May also need a slightly longer timeout, as I BELIEVE that systemd 90 second default is per service, so if it's shutting down multiple services, it can definitely take longer than that. I'd think 2 or maybe even 5 minutes, since this isn't going to handle the failure case (where reboot doesn't happen.) Regardless, I'd DEFINITELY suggest logging an error if the timeout expires (included in suggestion above... rewrite as needed) so the user knows things are off the rails.

Reply:
makes sense, thanks i missed there's a sleep on the main loop